### ISSUE TITLE
Tor / I2P node: How serious are these log messages? ("Unable to send transaction(s) to (tor|i2p) - no suitable outbound connections at height X", "Incoming connections disabled, enable them for full connectivity")

### ISSUE THREAD

**Issue by xanoni** [monero-project/monero#7863]
I keep seeing below messages, should I be concerned or are these just IP-artifacts that don't matter in case of Tor / I2P nodes (see configuration below)?

1. `I Incoming connections disabled, enable them for full connectivity`
2. `W Unable to send transaction(s) to i2p - no suitable outbound connections at height Y`
3. `W Unable to send transaction(s) to tor - no suitable outbound connections at height X`

I see (2) and (3) maybe 10-25x per hour, whereas (1) is less frequent at ~2x per hour.

**Settings / Version:**

`Monero 'Oxygen Orion' (v0.17.2.0-release)` with the below key settings:
```
tx-proxy=i2p,127.0.0.1:4447,25
tx-proxy=tor,127.0.0.1:9050,25

anonymous-inbound=XYZ.b32.i2p:${I2P_EXT_PORT},127.0.0.1:18080,25 # `i2pd` tunnel was defined manually
anonymous-inbound=XYZ.onion:${TOR_EXT_PORT},127.0.0.1:18080,25

# settings equivalent to the below also used for RPC 
p2p-ignore-ipv4=1
p2p-use-ipv6=0
p2p-bind-ip=127.0.0.1
p2p-bind-port=18080

hide-my-port=1
public-node=0
no-igd=1
allow-local-ip=0
no-zmq=1
```

EDIT1: `torsocks` / `proxychains` / `torify` are NOT used.
EDIT2: Another observation, not sure if related — I see only very few Tor and I2P nodes among my connected peers. Which setting should be tweaked to increase the ratio?

**Comment by xanoni** [comment]
I guess at least the first one is serious ... 

```
$ xmrd status
2021-08-16 03:15:34.391 I Monero 'Oxygen Orion' (v0.17.2.0-release)
Height: 2428087/2428087 (100.0%) on mainnet, not mining, net hash 2.69 GH/s, v14, 64(out)+0(in) connections, uptime X
```

Which setting is this, and can I control on which networks I want to accept incoming connections? E.g., only Tor, only I2P, only IPv6, etc.?

**Comment by xanoni** [comment]
No clue. Can someone say if my hypothesis is correct, i.e., that `--hide-my-port` needs to be removed for `anonymous-inbound` to work? Or is it something else? Still no peers after more than a day.

**Comment by xanoni** [comment]
Alright, the removal of `--hide-my-port` actually seems to have fixed it. The confusion was caused by the fact that the `status` RPC command only shows IPv4 / IPv6 peers.

```bash
$ monerod status
2021-08-19 12:21:44.344 I Monero 'Oxygen Orion' (v0.17.2.0-release)
Height: 2430372/2430372 (100.0%) on mainnet, not mining, net hash 2.63 GH/s, v14, 0(out)+0(in) connections, uptime 1d 5h 14m 41s
```

I also realized that the `in_peers` and `out_peers` config lines only affect IPv4/IPv6 connections and are NOT inclusive of the Tor and I2P connection count defined via `--anonymous-inbound` and `--tx-proxy`. In the above, I set both to 0. Only the Tor and I2P connections (which are now plenty) remain when I check `monerod print_cn`.

The documentation doesn't explain all these things very well.
